### PR TITLE
Migrate `constants` to C++

### LINKED ISSUE #998: Migrate `constants.py` to C++

### PR DESCRIPTION
**Pull request by feltech**
## Description

Closes #998

- [x] I have updated the release notes.
- [ ] ~~I have updated all relevant user documentation.~~

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ v1.0.0-alpha.xx
   module. These have been replaced by domain-specific traits and
   specifications defined in their own repositories, such as
   [OpenAssetIO-MediaCreation](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation).
+  Renamed remaining constants to have an `InfoKey_` prefix and migrated
```
Comment (foundrytom) [inline RELEASE_NOTES.md:33]: [fix] Can we add a `### Deprecated` section below breaking changes and note that the old prefixes will be removed in a future release?

File: src/openassetio-python/cmodule/src/constantsBinding.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "_openassetio.hpp"
+
+void registerConstants(const py::module_ &mod) {
+  mod.attr("kInfoKey_Icon") = openassetio::constants::kInfoKey_Icon;
+  mod.attr("kInfoKey_SmallIcon") = openassetio::constants::kInfoKey_SmallIcon;
+  mod.attr("kInfoKey_EntityReferencesMatchPrefix") =
+      openassetio::constants::kInfoKey_EntityReferencesMatchPrefix;
+  // TODO(DF): @deprecated
+  mod.attr("kField_EntityReferencesMatchPrefix") =
```
Comment (foundrytom) [inline src/openassetio-python/cmodule/src/constantsBinding.cpp:16]: [fix] As its so trivial, and I noticed we do document the icon fields, can we add aliases for those too please?

File: RELEASE_NOTES.md (hunk 1 above)
Comment (foundrytom) [inline RELEASE_NOTES.md:33]: [fix] Can we update documentation and code references to the new names?

**Review by foundrytom — CHANGES_REQUESTED** [review]

File: RELEASE_NOTES.md (hunk 1 above)
Comment (feltech) [reply RELEASE_NOTES.md]: Added deprecated section in db9870c5

File: src/openassetio-python/cmodule/src/constantsBinding.cpp (hunk 2 above)
Comment (feltech) [reply src/openassetio-python/cmodule/src/constantsBinding.cpp:18]: Reintroduced in 4a70aeec

File: RELEASE_NOTES.md (hunk 1 above)
Comment (feltech) [reply RELEASE_NOTES.md]: Whoops! Loads of places missed. Fixed in 165dd87f

**Review by foundrytom — APPROVED** [review]

**Pull request merged**